Bio/SeqUtils/__init__.py

```diff
@@ -467,6 +468,69 @@ def six_frame_translations(seq, genetic_code=1):
     return res
 
 
+def count_kmers(seq, k, circular=False, ordered=True):
+    """Count how many times each kmer appears in seq.
+
+    Counts instances of each substring of size k (kmers) in seq. Returns a
+    python collections.Counter object, with the kmers as keys and the number
+    of occurence as values. Overlaping k-mers, such as "ARA" in "ARARA" will be
+    counted twice.
+
+    Arguments:
+     - seq: String or Biopython sequence object.
+     - k: integer with size of kmer to be counted.
+     - circular: bool. Is the molecule circular?
+     - ordered: bool. If False will count "AGC" and "CGA" as the same kmer. Will
+                also count palindromes twice.
+
+
+    >>> from Bio.SeqUtils import count_kmers
+    >>> counts = count_kmers("RAIVMGARARAGRWKGARAL", 3)
+    >>> print(f"Number of times 'GAR' appears in seq: {counts['GAR']:02d}")
+    Number of times 'GAR' appears in seq: 02
+    >>> print(f"Number of times 'ARA' appears in seq: {counts['ARA']:02d}")
+    Number of times 'ARA' appears in seq: 03
+
+    Counter objects return 0 instead of throwing errors if kmer is not present.
+
+    >>> print(f"Number of times 'LRA' appears in seq: {counts['LRA']:02d}")
+    Number of times 'LRA' appears in seq: 00
+
+    If ordered = False, wil count the kmers in the reverse order as the same.
+
+    >>> counts = count_kmers("RAIVMGARARAGRWKGARAL", 3, ordered = False)
+    >>> print(f"Number of times 'GAR' appears in seq: {counts['GAR']:02d}")
+    Number of times 'GAR' appears in seq: 03
+    >>> print(f"Number of times 'RAG' appears in seq: {counts['RAG']:02d}")
+    Number of times 'RAG' appears in seq: 03
+    >>> print(f"Number of times 'ARA' appears in seq: {counts['ARA']:02d}")
+    Number of times 'ARA' appears in seq: 06
+
+    If circular = True, will wrap around the end of the sequence as well.
+
+    >>> counts = count_kmers("RAIVMGARARAGRWKGARAL",3, circular = True)
+    >>> print(f"Number of times 'ALR' appears in seq: {counts['ALR']:02d}")
+    Number of times 'ALR' appears in seq: 01
+
+    """
+    kmer_counts = Counter()
+
+    for i in range(len(seq) - k + 1):
+        kmer = seq[i : i + k]
+        kmer_counts[kmer] += 1
+        if not ordered:
```

Comment:
See comment above about the naming and use of the "ordered" option, but repeating the if statement in the loop seems to me a bad idea for performance.

(I am expecting people will try this function on whole genome scale datasets, not just short proteins)